src/parser/bison_parser.y

```diff
@@ -978,8 +987,35 @@ comp_expr : operand '=' operand { $$ = Expr::makeOpBinary($1, kOpEquals, $3); }
 | operand LESSEQ operand { $$ = Expr::makeOpBinary($1, kOpLessEq, $3); }
 | operand GREATEREQ operand { $$ = Expr::makeOpBinary($1, kOpGreaterEq, $3); };
 
-function_expr : IDENTIFIER '(' ')' { $$ = Expr::makeFunctionRef($1, new std::vector<Expr*>(), false); }
-| IDENTIFIER '(' opt_distinct expr_list ')' { $$ = Expr::makeFunctionRef($1, $4, $3); };
+function_expr : IDENTIFIER '(' ')' opt_window { $$ = Expr::makeFunctionRef($1, new std::vector<Expr*>(), false, $4); }
```

Comment:
I would split the general function expression (which can occur also in the where clause) and the specific window function, which is only allowed in the SELECT and ORDER BY clause and comes with a special syntax, requiring the keyword "OVER".

I think, this would, in particular, benefical for future extensions, such as advanced aggregate expression.

Reply:
As discussed today, we could add a comment:
`// function_expr is used for window functions, aggregate expressions, and functions calls because we run into shift/reduce conflicts when splitting them`